### PR TITLE
Fix bug while roping players, fix #1766

### LINKED ISSUE #1766: Teleport use
Use ladder with pz tile is possible to use for skull players

### PR DESCRIPTION
**Pull request by wgrriffel**

### CODE REVIEW

File: data/actions/scripts/other/teleport.lua [hunk 1]
```diff
@@ -5,6 +5,11 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	else
 		fromPosition.z = fromPosition.z + 1
 	end
+	local t = Tile(fromPosition)
+	if t:hasFlag(TILESTATE_PROTECTIONZONE) and player:isPzLocked() then
```
Comment (marksamman) [inline data/actions/scripts/other/teleport.lua]: Change this, and the line above, to:
```
if player:isPzLocked() and Tile(fromPosition):hasFlag(TILESTATE_PROTECTIONZONE) then
```

Comment (marksamman) [inline data/actions/scripts/other/teleport.lua]: Apply the same change in rope.lua too.

**Comment by wgrriffel** [comment]
anything else? let me know then i can squash the commits.

**Comment by marksamman** [comment]
Thanks!

**Pull request merged**